www/source/blog/2017-08-09-Habitat-plans-for-net-full-iis-apps.md

```diff
@@ -0,0 +1,301 @@
+
+However, today, the world's .NET applications run on the .NET full framework and ASP.NET applications run inside IIS and not the light weight [Kestrel](https://github.com/aspnet/KestrelHttpServer) web server. How do we build and run these inside of Habitat? How would we create packages for the .NET full framework or IIS like we do for Node or nginx? These are truly challenging scenarios because the .NET full framework needs to reside in a spcific location on disk and IIS is a feature of the Windows operating system.
+
+This post will walk you through how to work around these challenges and create a plan that builds and runs an ASP.NET full framework application inside of an IIS application pool.
+
+You can find a complete github repo of this application along with its Habitat plan, hooks and config [here](https://github.com/habitat-sh/habitat-aspnet-full).
+
+## .NET full framework and IIS as Infrastructure
+
+First and foremost, we will not be creating habitat packages for the .NET framework or IIS. We are going to let our configuration management or provisioning system of choice ensure that the .NET framework is installed and that the appropriate IIS features are enabled. When we author our Habitat plan, we will assume that the .NET framework and IIS pieces are already in place. Our plan will focus on our individual .NET application. It will build the application on any machine where the .NET 4.0 clr is installed, it will setup the IIS application pool and web site, and deploy our app binaries and web artifacts to our nodes where the Habitat supervisor will ensure it is running and update it as updates are available.
+
+So lets get started!
+
+## An ASP.NET application
+
+The first thing we need is an actual ASP.NET application. You may already have one and if so, thats great. Because I don't right now (but I did spend 10 years of my career building them), I'm gonna cheat. I'll open [Visual Studio 2017](https://www.visualstudio.com/) and select file/new/project and then create the stock ASP.NET MVC application template. Earlier Visual Studio like 2015 should work too.
+
+![file-new-project](media/2017-08-29-Fullfx/new_proj.png)
+
+Finally, and mainly to illustrate how to adjust our Habitat plan to target different .NET framework versions, we will target version 4.7.
+
+![target-framework](media/2017-08-29-Fullfx/target.png)
+
+## Building with a plan.ps1
+
+Right in Visual Studio I'll add a habitat folder to my project and then right click on that folder and select Add/New Item. Because I have the [Powershell Tools Visual Studio Extension](https://poshtools.com/), I have the option to add a powershell script and I will add a plan.ps1 file.
```

Comment:
If "habitat folder" refers to a literal directory named "habitat", we should put "habitat" in backticks; otherwise Habitat should be capitalized.

"powershell" => "PowerShell"

"plan.ps1" => `plan.ps1`